flashback: flashback cluster support tso (#15608)
pingcap/docs: sql-statements/sql-statement-flashback-to-timestamp.md

---
title: FLASHBACK CLUSTER
summary: Learn the usage of FLASHBACK CLUSTER in TiDB databases.
aliases: ['/tidb/v6.5/sql-statement-flashback-to-timestamp']
---

# FLASHBACK CLUSTER

> TiDB v6.4.0 introduces the `FLASHBACK CLUSTER TO TIMESTAMP` syntax. You can use it to restore a cluster to a specific point in time. When specifying the timestamp, you can set a datetime value or use a time function. The format of datetime is like "2016-10-08 16:45:26.999", with millisecond as the minimum time unit, but in most cases, the time unit of second is enough for specifying a datetime, such as "2016-10-08 16:45:26".

Author: ```suggestion
TiDB v6.4.0 introduces the `FLASHBACK CLUSTER TO TIMESTAMP` syntax. You can use it to restore a cluster to a specific point in time. When specifying the timestamp, you can either set a datetime value or use a time function. The format of datetime is like "2016-10-08 16:45:26.999", with the millisecond as the minimum time unit, but in most cases, specifying the timestamp with a second as the time unit, such as "2016-10-08 16:45:26", is sufficient.
```

> TiDB v6.4.0 introduces the `FLASHBACK CLUSTER TO TIMESTAMP` syntax. You can use it to restore a cluster to a specific point in time. When specifying the timestamp, you can either set a datetime value or use a time function. The format of datetime is like "2016-10-08 16:45:26.999", with millisecond as the minimum time unit, but in most cases, specifying the timestamp with second as the time unit is sufficient, such as "2016-10-08 16:45:26".

Reviewer: ```suggestion
TiDB v6.4.0 introduces the `FLASHBACK CLUSTER TO TIMESTAMP` syntax. You can use it to restore a cluster to a specific point in time. When specifying the timestamp, you can either set a datetime value or use a time function. The format of datetime is like "2016-10-08 16:45:26.999", with millisecond as the minimum time unit. But in most cases, specifying the timestamp with second as the time unit is sufficient, for example, "2016-10-08 16:45:26".
```

> TiDB v6.4.0 introduces the `FLASHBACK CLUSTER TO TIMESTAMP` syntax. You can use it to restore a cluster to a specific point in time. When specifying the timestamp, you can either set a datetime value or use a time function. The format of datetime is like "2016-10-08 16:45:26.999", with millisecond as the minimum time unit. But in most cases, specifying the timestamp with second as the time unit is sufficient, for example, "2016-10-08 16:45:26".

Author: ```suggestion
TiDB v6.4.0 introduces the `FLASHBACK CLUSTER TO TIMESTAMP` syntax. You can use it to restore a cluster to a specific point in time. When specifying the timestamp, you can either set a datetime value or use a time function. The format of datetime is like `'2016-10-08 16:45:26.999'`, with millisecond as the minimum time unit. But in most cases, specifying the timestamp with second as the time unit is sufficient, for example, `'2016-10-08 16:45:26'`.
```

> Starting from v6.5.6, TiDB introduces the `FLASHBACK CLUSTER TO TSO` syntax. With this syntax, you can use [TSO](/tso.md) to specify a more accurate recovery point in time, which brings more flexibility in data recovery.

Author: ```suggestion
Starting from v6.5.6, TiDB provides you with a new option by introducing the `FLASHBACK CLUSTER TO TSO` syntax. With this syntax, you can use [TSO](/tso.md) to specify a more accurate recovery point in time, which brings more flexibility in data recovery.
```

Empty set (0.00 sec)
```

> The following example shows how to flashback a cluster to a specific TSO to accurately restore mistakenly deleted data:

Author: ```suggestion
The following example shows how to flashback a cluster to a specific TSO to precisely restore mistakenly deleted data:
```

> The following example shows how to flashback a cluster to a specific TSO to accurately restore mistakenly deleted data::

Author: ```suggestion
The following example shows how to flashback a cluster to a specific TSO to accurately restore mistakenly deleted data:
```

```sql
mysql> INSERT INTO t VALUES (1);
Query OK, 1 row affected (0.02 sec)

mysql> SELECT * FROM t;
+------+
| a    |
+------+
|    1 |
+------+
1 row in set (0.01 sec)

> mysql> begin;

Reviewer: ```suggestion
mysql> BEGIN;
```

Query OK, 0 rows affected (0.00 sec)

> mysql> select @@tidb_current_ts;  -- Get the current TSO

Reviewer: ```suggestion
mysql> SELECT @@tidb_current_ts;  -- Get the current TSO
```

> mysql> select @@tidb_current_ts;   -- Get the current TSO

Author: ```suggestion
mysql> select @@tidb_current_ts;  -- Get the current TSO
```

+--------------------+
| @@tidb_current_ts  |
+--------------------+
| 446113975683252225 |
+--------------------+
1 row in set (0.00 sec)

> mysql> rollback;

Reviewer: ```suggestion
mysql> ROLLBACK;
```
